Update Fortigate RestAPI procedure (#2732)
centreon/centreon-documentation: i18n/fr/docusaurus-plugin-content-docs-pp/current/integrations/plugin-packs/procedures/network-fortinet-fortigate-restapi.md

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Contenu du pack

### Modèles

Le connecteur de supervision **Fortinet Fortigate Rest API** apporte un modèle d'hôte :

* **Net-Fortinet-Fortigate-Restapi-custom**

Le connecteur apporte les modèles de service suivants
(classés selon le modèle d'hôte auquel ils sont rattachés) :

<Tabs groupId="sync">
<TabItem value="Net-Fortinet-Fortigate-Restapi-custom" label="Net-Fortinet-Fortigate-Restapi-custom">

| Alias    | Modèle de service                              | Description                              |
|:---------|:-----------------------------------------------|:-----------------------------------------|

> | Health   | Net-Fortinet-Fortigate-Health-Restapi-custom   | Contrôle l'état des santé du firewall    |

Reviewer: ```suggestion
| Health   | Net-Fortinet-Fortigate-Health-Restapi-custom   | Contrôle l'état de santé du firewall    |
```

| Licenses | Net-Fortinet-Fortigate-Licenses-Restapi-custom | Contrôle le statut des licences          |
| System   | Net-Fortinet-Fortigate-System-Restapi-custom   | Contrôle l'utilisation système des VDOM  |

> Les services listés ci-dessus sont créés automatiquement lorsque le modèle d'hôte **Net-Fortinet-Fortigate-Restapi-custom** est utilisé.

</TabItem>
<TabItem value="Non rattachés à un modèle d'hôte" label="Non rattachés à un modèle d'hôte">

| Alias | Modèle de service                        | Description                                            |
|:------|:-----------------------------------------|:-------------------------------------------------------|

> | Ha    | Net-Fortinet-Fortigate-Ha-Restapi-custom | Contrôle l'utilisation système des members du cluster  |

Reviewer: ```suggestion
| Ha    | Net-Fortinet-Fortigate-Ha-Restapi-custom | Contrôle l'utilisation système des membres du cluster  |
```

| WARNINGHEALTH  | Define the conditions to match for the status to be WARNING. You can use the following variables: %{status}, %{name}                                       |                         |             |
| EXTRAOPTIONS   | Any extra option you may want to add to the command (E.g. a --verbose flag). Toutes les options sont listées [ici](#options-disponibles)                                                        | --verbose               |             |

</TabItem>
<TabItem value="Licenses" label="Licenses">

| Macro           | Description                                                                                                                                                | Valeur par défaut       | Obligatoire |
|:----------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------|:------------------------|:-----------:|
| FILTERNAME      | Filter licenses by name (can be a regexp)                                                                                                                  |                         |             |

> | UNIT            | Select the unit for expires threshold. May be 's' for seconds,'m' for minutes, 'h' for hours, 'd' for days, 'w' for weeks. Default is seconds              |                         |             |

Reviewer: ```suggestion
| UNIT            | Select the time unit for thresholds. May be 's' for seconds,'m' for minutes, 'h' for hours, 'd' for days, 'w' for weeks. Default is seconds              |                         |             |
```

</TabItem>
</Tabs>

3. [Déployez la configuration](/docs/monitoring/monitoring-servers/deploying-a-configuration). Le service apparaît dans la liste des services supervisés, et dans la page **Statut des ressources**. La commande envoyée par le connecteur est indiquée dans le panneau de détails du service : celle-ci montre les valeurs des macros.

## Comment puis-je tester le plugin et que signifient les options des commandes ?

Une fois le plugin installé, vous pouvez tester celui-ci directement en ligne
de commande depuis votre collecteur Centreon en vous connectant avec
l'utilisateur **centreon-engine** (`su - centreon-engine`). Vous pouvez tester
que le connecteur arrive bien à superviser une ressource en utilisant une commande
telle que celle-ci (remplacez les valeurs d'exemple par les vôtres) :

```bash
/usr/lib/centreon/plugins/centreon_fortinet_fortigate_restapi.pl \
	--plugin=network::fortinet::fortigate::restapi::plugin \
	--mode=system \
	--hostname='10.0.0.1' \
	--port='443' \
	--proto='https' \

> 	--access-token=''  \

Reviewer: token?

> 	--access-token=''  \

Author: ```suggestion
	--access-token='xxxxxx'  \
```

	--filter-vdom='' \
	--warning-cpu-utilization='' \
	--critical-cpu-utilization='' \
	--warning-sessions-active='' \
	--critical-sessions-active='' \
	--warning-memory-usage='' \
	--critical-memory-usage='' \
	--verbose
```

La commande devrait retourner un message de sortie similaire à :

```bash

> OK: cpu load: 87 % memory used: 15 % active sessions: 99 | '*vdoms*~cpu.utilization.percentage'=87%;;;0;100'*vdoms*~memory.usage.percentage'=15%;;;0;100'*vdoms*~sessions.active.count'=99;;;0;

Reviewer: output is very different

> OK: cpu load: 87 % memory used: 15 % active sessions: 99 | '*vdoms*~cpu.utilization.percentage'=87%;;;0;100'*vdoms*~memory.usage.percentage'=15%;;;0;100'*vdoms*~sessions.active.count'=99;;;0;

Author: ```suggestion
OK: All vdom systems are ok | 'ABS#cpu.utilization.percentage'=0.00%;;;0;100 'ABS#memory.usage.percentage'=0.00%;;;0;100 'ABS#sessions.active.count'=155;;;0; 'ADV#cpu.utilization.percentage'=0.00%;;;0;100 'ADV#memory.usage.percentage'=1.00%;;;0;100 'ADV#sessions.active.count'=553;;;0; 'BGN#cpu.utilization.percentage'=0.00%;;;0;100 'BGN#memory.usage.percentage'=0.00%;;;0;100 'BGN#sessions.active.count'=244;;;0; 'LHE#cpu.utilization.percentage'=0.00%;;;0;100 'LHE#memory.usage.percentage'=0.00%;;;0;100 'LHE#sessions.active.count'=100;;;0; 'MED#cpu.utilization.percentage'=3.00%;;;0;100 'MED#memory.usage.percentage'=11.00%;;;0;100 'MED#sessions.active.count'=6280;;;0; 'MIC#cpu.utilization.percentage'=0.00%;;;0;100 'MIC#memory.usage.percentage'=5.00%;;;0;100 'MIC#sessions.active.count'=3244;;;0; 'MLC#cpu.utilization.percentage'=0.00%;;;0;100 'MLC#memory.usage.percentage'=0.00%;;;0;100 'MLC#sessions.active.count'=431;;;0; 'PRN#cpu.utilization.percentage'=0.00%;;;0;100 'PRN#memory.usage.percentage'=0.00%;;;0;100 'PRN#sessions.active.count'=0;;;0; 'SSTRN#cpu.utilization.percentage'=5.00%;;;0;100 'SSTRN#memory.usage.percentage'=12.00%;;;0;100 'SSTRN#sessions.active.count'=6559;;;0; 'root#cpu.utilization.percentage'=2.00%;;;0;100 'root#memory.usage.percentage'=4.00%;;;0;100 'root#sessions.active.count'=228;;;0;
```
